microsoft/PowerToys
# Powertoyrun confirm

## Summary of the Pull Request
Adds new feature #12293

**What is this about:**
Request was made to have a confirmation dialog used for system commands such as shutdown, log off, sleep, hibernate,and restart.

**What is include in the PR:** 
Changes 2 about 10 files to include a setting in the settings screen to enable/disable this feature.

**How does someone test / validate:** 
Build and start this version of powertoys.  Go to the settings screen and select PowerToys Run in the left column,  Now scroll all the say down and click on Windows System Commands.  Turn on the option "Confirm System Actions".  Press the key combination that opens PowerToy Run - ALT + SPACE by default.  Type in a system command such as "Shut Down'.  Verify that you get a confirmation dialog.  Try telling it "No", verify nothing happens.  Launch the PowerToy again, Type a System command again, but this time click "Yes" and veryify that the operation completes as expected.  Repeat the process for each of "Shut Down", "Sleep", "Hibernate", "Lock", and "Log Out"

Next, go back to the settings and turn off the option to confirm system commands.  Verify that they all work as expected with no confirmation box.

## Quality Checklist

- [x] **Linked issue:** #12293
- [x] **Tests:** All pass, no new tests added
- [x] **Installer:** No Changes required to installer
- [X] **Localization:** No changes that need to be localized

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
